lib/ui/window/platform_message_response_dart.cc

```diff
@@ -19,30 +19,40 @@ namespace {
 // Avoid copying the contents of messages beyond a certain size.
 const int kMessageCopyThreshold = 1000;
 
+class DataWrapper {
+ public:
+  DataWrapper(void* data) { data_ = data; }
+
+  ~DataWrapper() { free(data_); }
+
+ private:
+  void* data_;
+};
+
 void MessageDataFinalizer(void* isolate_callback_data,
                           Dart_WeakPersistentHandle handle,
                           void* peer) {
-  std::vector<uint8_t>* data = reinterpret_cast<std::vector<uint8_t>*>(peer);
+  DataWrapper* data = reinterpret_cast<DataWrapper*>(peer);
   delete data;
 }
 
-Dart_Handle WrapByteData(std::vector<uint8_t> data) {
-  if (data.size() < kMessageCopyThreshold) {
+Dart_Handle WrapByteData(std::unique_ptr<fml::Mapping> mapping) {
+  size_t size = mapping->GetSize();
+  if (size < kMessageCopyThreshold) {
+    std::vector<uint8_t> data(size);
+    memcpy(data.data(), mapping->GetMapping(), size);
     return ToByteData(data);
   } else {
-    std::vector<uint8_t>* heap_data = new std::vector<uint8_t>(std::move(data));
-    return Dart_NewExternalTypedDataWithFinalizer(
-        Dart_TypedData_kByteData, heap_data->data(), heap_data->size(),
-        heap_data, heap_data->size(), MessageDataFinalizer);
+    void* data = malloc(size);
+    memset(data, 0, size);
```

Comment:
The `memset` seems to be unnecessary?

Reply:
> It would be nice to add a microbenchmark to test `PlatformMessageResponseDart` with a 3MB asset. See, e.g., the benchmark in https://github.com/flutter/engine/blob/master/shell/common/shell_benchmarks.cc

PlatformMessageResponseDart only post a ui task, and this ui task will consume time when execute

Reply:
> The `memset` seems to be unnecessary?

Yes, just according to the logical of std::vector